### PR TITLE
Circle CI Config Cleanup

### PR DESCRIPTION
**Pull request by macdrevx**
## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'

### Summary of changes

* Configures a linter (yamllint) for yaml files and resolves lint issues
* Applies some minor other refactoring to the Circle CI config
* Deletes unused xcframework target in the Makefile
* Deletes some unreferenced files

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -36,15 +37,11 @@ comment:
...
-     carryforward: false
+    carryforward: false
   MapboxMapsTestsWithHost:
     paths:
       - Mapbox/
-     carryforward: false
-  MapboxMapsTestsWithHost:
-    paths:
-      - Mapbox/
-     carryforward: false
```
Comment (macdrevx) [inline codecov.yml:50]: This was a duplicate dictionary key

File: scripts/release/packager/project.yml [hunk 2]
```diff
@@ -8,24 +9,23 @@ targets:
...
         INSTALL_PATH: $(LOCAL_LIBRARY_DIR)/Frameworks
-        SKIP_INSTALL: NO
+        SKIP_INSTALL: "NO"
         TARGETED_DEVICE_FAMILY: 1,2
         DYLIB_COMPATIBILITY_VERSION: 1
         DYLIB_CURRENT_VERSION: 3
         LD_RUNPATH_SEARCH_PATHS: $(inherited) @executable_path/Frameworks @loader_path/Frameworks
         MACH_O_TYPE: mh_dylib
-        DEFINES_MODULE: YES
-        PRODUCT_BUNDLE_IDENTIFIER: com.mapbox.MapboxMaps
```
Comment (macdrevx) [inline scripts/release/packager/project.yml:18]: Duplicate dictionary key

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,700 +1,655 @@
...
-            xcode:
-                type: string
-                default: "12.2.0"
-        macos:
-            xcode: << parameters.xcode >>
-        environment:
-            HOMEBREW_NO_AUTO_UPDATE: 1
-            HOMEBREW_NO_INSTALL_CLEANUP: 1
-
-# ======================================================================================================================
```
Comment (julianrex) [inline .circleci/config.yml:23]: I don't think we should remove this - it has been useful in the past to test against multiple versions of Xcode using the matrix functionality (e.g. when we want to test with current stable and beta versions).

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,700 +1,655 @@
...
+          command: make symbolicate SCHEME=<< parameters.scheme >> APP_NAME=<< parameters.app-name >> CONFIGURATION=Release
+          when: always
+      - run:
+          name: Converting and uploading coverage
+          command: make device-update-codecov-with-profdata SCHEME=<< parameters.scheme >> APP_NAME=<< parameters.app-name >> CONFIGURATION=Release
+      - store-device-farm-artifacts
+      - store-logs
+      - report-failure:
+          report_failure: << parameters.report_failure >>
+          message: "`<< parameters.scheme >>` device tests"
```
Comment (julianrex) [inline .circleci/config.yml:296]: If you remove the back-ticks here, we can close https://github.com/mapbox/mapbox-maps-ios/pull/171 😄

**Review by julianrex — APPROVED** [review]
Looks great - thanks. Just a couple of comments

File: .circleci/config.yml (hunk 3 above)
Comment (macdrevx) [reply .circleci/config.yml:23]: okay, will revert

Comment (macdrevx) [reply .circleci/config.yml:23]: [This](https://github.com/mapbox/mapbox-maps-ios/pull/175/commits/d42c11eda4583684c2d2771cfbe00ca900c45037) should be sufficient, yeah?

File: .circleci/config.yml (hunk 4 above)
Comment (macdrevx) [reply .circleci/config.yml]: Updated

File: .circleci/config.yml (hunk 3 above)
Comment (julianrex) [reply .circleci/config.yml:23]: Yep, and re-add the executor section.

Comment (julianrex) [reply .circleci/config.yml:23]: Oh I see. Yeah, let's keep that.

**Pull request merged**